model/dyn_core.F90

```diff
@@ -288,6 +300,10 @@ subroutine dyn_core(npx, npy, npz, ng, sphum, nq, bdt, n_map, n_split, zvir, cp,
     logical used
     real :: split_timestep_bc
 
+!The following is for SA-3D-TKE
+    real:: tke(bd%isd:bd%ied,bd%jsd:bd%jed,npz)
+    integer :: ntke
```

Comment:
In FV3 (and FMS in general) the convention for tracer indices is to use the same name as the tracer field; so I would define `sgs_tke` instead of `ntke`.